cmd/flux/push_artifact.go

```diff
@@ -129,8 +129,18 @@ func pushArtifactCmdRun(cmd *cobra.Command, args []string) error {
 		return err
 	}
 
-	if _, err := os.Stat(pushArtifactArgs.path); err != nil {
-		return fmt.Errorf("invalid path '%s', must point to an existing directory or file", buildArtifactArgs.path)
+	path := pushArtifactArgs.path
+	if pushArtifactArgs.path == "-" {
+		path, err = saveStdinToFile()
```

Comment:
`saveStdinToFile` is leaving the temp file opened. We need to close it before returning, otherwise `os.Remove()` will fail in some systems.

Please add a test for the stdin feature.